### DISCUSSION TITLE
我有 build macOS 的執行檔

### DISCUSSION THREAD

**Discussion by mosluce** [GpointChen/FFXIVChnTextPatch-GP#202] [Ideas]
要怎麼提供比較方便呢？

**Comment by mosluce** [comment]
release 
https://github.com/mosluce/FFXIVChnTextPatch-GP/releases/tag/latest

**Comment by onlineid** [comment]
請問M1的Macbook可以適用嗎？

**Comment by mosluce** [comment]
@onlineid 
可以試試看，我目前還沒有 M1 的 mac QQ

**Comment by Fredpkc** [comment]
Hi 我可以成功開啟Patch黨, 但請問mac FFXIV的預設路徑在哪？我選擇library/application support/Final Fantasy XIV ONLINE會顯示路徑錯誤

**Comment by Fredpkc** [comment]
![截圖 2022-07-28 下午10 48 03](https://user-images.githubusercontent.com/110171041/181566043-a1f7eb31-af69-4b44-b7f6-5d8acc719a8f.png)
![截圖 2022-07-28 下午10 48 22](https://user-images.githubusercontent.com/110171041/181566166-635fe99b-a4f0-4713-839d-2614656a0734.png)

**Comment by mosluce** [comment]
繼續點進去，路進應該是這樣
~/Library/Application\ Support/FINAL\ FANTASY\ XIV\
ONLINE/Bottles/published_Final_Fantasy/drive_c/Program\ Files\
\(x86\)/SquareEnix/FINAL\ FANTASY\ XIV\ -\ A\ Realm\ Reborn

Fredpkc ***@***.***> 於 2022年7月28日 週四 晚上10:48寫道：

**Comment by Fredpkc** [comment]
感謝...但在實際漢化的時候有Error:
Exception in thread "Thread-1" java.lang.NoSuchMethodError: java.nio.ByteBuffer.position(I)Ljava/nio/ByteBuffer;
	at name.yumao.ffxiv.chn.builder.TexBlockBuilder.buildBlock(TexBlockBuilder.java:38)
	at name.yumao.ffxiv.chn.replace.ReplaceFont.replace(ReplaceFont.java:70)
	at name.yumao.ffxiv.chn.thread.ReplaceThread.run(ReplaceThread.java:46)
	at java.lang.Thread.run(Thread.java:750)

**Comment by Fredpkc** [comment]
![截圖 2022-07-28 下午11 43 05](https://user-images.githubusercontent.com/110171041/181580787-1841c248-a64c-477c-8ff3-dc06757a5c2a.png)

**Comment by Fredpkc** [comment]
Java版本: 
java version "1.8.0_341"
Java(TM) SE Runtime Environment (build 1.8.0_341-b10)
Java HotSpot(TM) 64-Bit Server VM (build 25.341-b10, mixed mode)

我的MBP是M1, 但我想應該沒有影響

**Comment by mosluce** [comment]
@Fredpkc 
我的工具要搭配原作者那一包使用喔，我這包沒有語系檔

**Comment by Fredpkc** [comment]
@mosluce 我是按照你提供的步驟：
到 https://github.com/GpointChen/FFXIVChnTextPatch-GP/releases 下載 FFXIVChnTextPatch_0706.rar 並解壓縮
下載 FFXIVChnTextPatch_0706_macos_intel.zip 解壓縮後會得到兩個資料夾和一個執行檔，搬到上面解壓縮同一資料夾中
雙擊 FFXIVChnTextPatchGP (沒有 .exe) 執行

這樣也沒有語系檔媽？

**Comment by mosluce** [comment]
FFXIVChnTextPatchGP 和 FFXIVChnTextPatchGP.exe 要在同一層喔

如果還是不行的話，我就不知道囉，我現在沒有 M1 的機器可以測試

**Comment by Fredpkc** [comment]
了解 謝謝你了

**Comment by stussykaki** [comment]
> 了解 謝謝你了

你好，小弟剛進坑的新手，爬文找到這裡，也是跟你一樣卡了在font6.tex那裡，請問大大現在有解決了嗎？

**Reply by hydai** [reply]
Hi @stussykaki 
這個問題在新的討論串中解決了，可以嘗試看看： https://github.com/GpointChen/FFXIVChnTextPatch-GP/discussions/342

**Comment by onlineid** [comment]
@mosluce 
成功中文化了，整個3.0都沒有遇到問題，特來致謝。辛苦了。

順便問一下，現在中文化質量最高的是用哪個方式呢？我現在用的是CSV的方式。但是遊戲語音還是英文，所以感覺有些翻譯還是不太準確。我知道這邊只是提供中文化的程式而不負責翻譯質量，只是希望可以借鑑前輩的經驗。

**Comment by Fredpkc** [comment]
> > 了解 謝謝你了
> 
> 你好，小弟剛進坑的新手，爬文找到這裡，也是跟你一樣卡了在font6.tex那裡，請問大大現在有解決了嗎？
 
沒有耶 我直接放棄  看英文了

**Comment by Fredpkc** [comment]
> @mosluce 成功中文化了，整個3.0都沒有遇到問題，特來致謝。辛苦了。
> 
> 順便問一下，現在中文化質量最高的是用哪個方式呢？我現在用的是CSV的方式。但是遊戲語音還是英文，所以感覺有些翻譯還是不太準確。我知道這邊只是提供中文化的程式而不負責翻譯質量，只是希望可以借鑑前輩的經驗。

大部分中文化是對岸的官方文本轉繁體過來的，質量已經很高了
我先前也是看英文玩，官方英文文本的順序跟一些對話與官方中日文差很多，並不是直接翻譯。估計文本是有重寫。
但中文文本就感覺得出來是日文翻譯過來而沒有重寫。
所有你才會覺得中文字幕跟英文語音對不上吧

**Comment by onlineid** [comment]
> > @mosluce 成功中文化了，整個3.0都沒有遇到問題，特來致謝。辛苦了。
> > 順便問一下，現在中文化質量最高的是用哪個方式呢？我現在用的是CSV的方式。但是遊戲語音還是英文，所以感覺有些翻譯還是不太準確。我知道這邊只是提供中文化的程式而不負責翻譯質量，只是希望可以借鑑前輩的經驗。
> 
> 大部分中文化是對岸的文本轉繁體過來的，質量已經很高了 我先前也是看英文玩，官方英文文本的順序跟一些對話與官方中日文差很多，並不是直接翻譯。估計文本是有重寫。 所有你才會覺得中文字幕跟英文語音對不上吧

原來如此，明白了。感謝。

**Reply by ClareTW** [reply]
請問一下，你是用M1 macbook成功的嗎？

**Reply by onlineid** [reply]
@ClareTW 是的。M1 MBP, Monterey 12.4

**Reply by Fredpkc** [reply]
我會卡在Font6.tex這...可以分享你的步驟嗎

**Reply by onlineid** [reply]
1. 安裝Homebrew
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

2. 安裝OpenJDK
`brew install openjdk`

3. 確認java路徑
`which java`
> /opt/homebrew/Cellar/openjdk/18.0.2/bin//java 

4. 確認當前Shell環境
`env|grep SHELL`
> SHELL=/bin/zsh

5. 把java路徑加入Shell環境變數
`echo "export PATH=\"/opt/homebrew/Cellar/openjdk/18.0.2/bin/:\$PATH\"" >> ~/.zshrc`

6. 確認5的操作結果
`cat ~/.zshrc`
> export PATH="/opt/homebrew/Cellar/openjdk/18.0.2/bin/:$PATH"

7. 令環境變數生效
`source ~/.zshrc`

8. 下載中文化補丁"FFXIVChnTextPatch-0706"
> https://github.com/GpointChen/FFXIVChnTextPatch-GP/releases

9. 下載完之後是"FFXIVChnTextPatch-0706.rar" (現在最新版本是0809，但是因爲樓主發佈的Mac程式是支援0706，我不確定是否支援0809)

10. 解壓縮"FFXIVChnTextPatch-0706.rar"，裏面有4個資料夾和兩個文檔
> -rw-r--r--@ 1 erickchan  staff  24908390 Feb 12 00:08 FFXIVChnTextPatchGP.exe
> -rw-r--r--@ 1 erickchan  staff      5326 Mar 19 09:46 README.md
> drwxr-xr-x@ 3 erickchan  staff        96 Jul  6 04:47 backup
> drwxr-xr-x@ 4 erickchan  staff       128 Aug 11  2021 conf
> drwxr-xr-x@ 8 erickchan  staff       256 Apr 16 11:10 jre
> drwxr-xr-x@ 5 erickchan  staff       160 Jun 28 14:51 resource

11. 然後再下載樓主的Mac中文化程式"FFXIVChnTextPatch_0706_macos_intel.zip"
> https://github.com/mosluce/FFXIVChnTextPatch-GP/releases/tag/latest

12. 解壓縮"FFXIVChnTextPatch_0706_macos_intel.zip"，裏面有兩個資料夾和一個文檔
> -rwxr-xr-x@  1 erickchan  staff   60 Jul 10 19:19 FFXIVChnTextPatchGP
> drwxr-xr-x@  5 erickchan  staff  160 Jul 10 19:12 bin
> drwxr-xr-x@ 14 erickchan  staff  448 Jul 10 18:02 lib

13. 把第12步的兩個資料夾和一個文檔複製到第9步解壓出來的"FFXIVChnTextPatch-0706"資料夾中
> -rwxr-xr-x@  1 erickchan  staff        60 Jul 10 19:19 FFXIVChnTextPatchGP
> -rw-r--r--@  1 erickchan  staff  24908390 Feb 12 00:08 FFXIVChnTextPatchGP.exe
> -rw-r--r--@  1 erickchan  staff      5326 Mar 19 09:46 README.md
> drwxr-xr-x@  3 erickchan  staff        96 Jul  6 04:47 backup
> drwxr-xr-x@  5 erickchan  staff       160 Jul 10 19:12 bin
> drwxr-xr-x@  4 erickchan  staff       128 Aug 11  2021 conf
> drwxr-xr-x@  8 erickchan  staff       256 Apr 16 11:10 jre
> drwxr-xr-x@ 14 erickchan  staff       448 Jul 10 18:02 lib
> drwxr-xr-x@  5 erickchan  staff       160 Jun 28 14:51 resource

14. 在Finder中運行"FFXIVChnTextPatchGP"
15. MacOS的安全檢測問"這個程式來自Internet，確認要執行嗎？"選擇"執行"。
16. 出現中文化程式界面，選擇"遊戲路徑"（請參考本Po上面樓主的解答選擇正確的遊戲路徑）
17. 檔案語言選擇"CSV"
18. 原始語言選擇"日文"
19. 點“確認”開始實行中文化

**Reply by ClareTW** [reply]
@onlineid 感謝你和樓主，再來試試看

**Reply by ClareTW** [reply]
感謝，成功了，我直接拿0809的版本去試，是可以的
而且我是用XIV on MAC下去玩的，遊戲路徑和樓主說的不同，選XIV on MAC下一層的ffiv那個就好

**Reply by onlineid** [reply]
是喔，晚點我也來更新到0809好了，感謝分享。

**Reply by ClareTW** [reply]
> 5. 把java路徑加入Shell環境變數
>    `vi ~/.zshrc`
> 
> > export PATH="/opt/homebrew/Cellar/openjdk/18.0.2/bin/:$PATH"
> 

加完環境變數要按
:wq
存檔離開

之前沒用過homebrew，又查了一下子才找到怎麼離開
分享給一樣遇到困難的人

**Reply by onlineid** [reply]
> 我還是無法執行...會卡在font.6, 我的java路徑與你們不同,在 /usr/bin/java

https://apple.stackexchange.com/questions/269588/how-does-usr-bin-java-work-on-mac
/usr/bin/java 是MacOS Monterey內建的Java，但這個Java也不是真正的Java，簡單說就是一個用來控制作業系統上安裝的所有不同版本的Java的類似於控制台的一個東西，除非你需要這樣的機能，不然我建議就不要碰它，用Homebrew裝一個OpenJDK來用就好。

**Reply by Fredpkc** [reply]
謝謝 改用echo加入SHELL後解決了

**Reply by stussykaki** [reply]
> 1. 安裝Homebrew
>    `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`
> 2. 安裝OpenJDK
>    `brew install openjdk`
> 3. 確認java路徑
>    `which java`
> 
> > /opt/homebrew/Cellar/openjdk/18.0.2/bin//java
> 
> 4. 確認當前Shell環境
>    `env|grep SHELL`
> 
> > SHELL=/bin/zsh
> 
> 5. 把java路徑加入Shell環境變數
>    `echo "export PATH=\"/opt/homebrew/Cellar/openjdk/18.0.2/bin/:\$PATH\"" >> ~/.zshrc`
> 6. 確認5的操作結果
>    `cat ~/.zshrc`
> 
> > export PATH="/opt/homebrew/Cellar/openjdk/18.0.2/bin/:$PATH"
> 
> 7. 令環境變數生效
>    `source ~/.zshrc`
> 8. 下載中文化補丁"FFXIVChnTextPatch-0706"
> 
> > https://github.com/GpointChen/FFXIVChnTextPatch-GP/releases
> 
> 9. 下載完之後是"FFXIVChnTextPatch-0706.rar" (現在最新版本是0809，但是因爲樓主發佈的Mac程式是支援0706，我不確定是否支援0809)
> 10. 解壓縮"FFXIVChnTextPatch-0706.rar"，裏面有4個資料夾和兩個文檔
> 
> > -rw-r--r--@ 1 erickchan  staff  24908390 Feb 12 00:08 FFXIVChnTextPatchGP.exe
> > -rw-r--r--@ 1 erickchan  staff      5326 Mar 19 09:46 README.md
> > drwxr-xr-x@ 3 erickchan  staff        96 Jul  6 04:47 backup
> > drwxr-xr-x@ 4 erickchan  staff       128 Aug 11  2021 conf
> > drwxr-xr-x@ 8 erickchan  staff       256 Apr 16 11:10 jre
> > drwxr-xr-x@ 5 erickchan  staff       160 Jun 28 14:51 resource
> 
> 11. 然後再下載樓主的Mac中文化程式"FFXIVChnTextPatch_0706_macos_intel.zip"
> 
> > https://github.com/mosluce/FFXIVChnTextPatch-GP/releases/tag/latest
> 
> 12. 解壓縮"FFXIVChnTextPatch_0706_macos_intel.zip"，裏面有兩個資料夾和一個文檔
> 
> > -rwxr-xr-x@  1 erickchan  staff   60 Jul 10 19:19 FFXIVChnTextPatchGP
> > drwxr-xr-x@  5 erickchan  staff  160 Jul 10 19:12 bin
> > drwxr-xr-x@ 14 erickchan  staff  448 Jul 10 18:02 lib
> 
> 13. 把第12步的兩個資料夾和一個文檔複製到第9步解壓出來的"FFXIVChnTextPatch-0706"資料夾中
> 
> > -rwxr-xr-x@  1 erickchan  staff        60 Jul 10 19:19 FFXIVChnTextPatchGP
> > -rw-r--r--@  1 erickchan  staff  24908390 Feb 12 00:08 FFXIVChnTextPatchGP.exe
> > -rw-r--r--@  1 erickchan  staff      5326 Mar 19 09:46 README.md
> > drwxr-xr-x@  3 erickchan  staff        96 Jul  6 04:47 backup
> > drwxr-xr-x@  5 erickchan  staff       160 Jul 10 19:12 bin
> > drwxr-xr-x@  4 erickchan  staff       128 Aug 11  2021 conf
> > drwxr-xr-x@  8 erickchan  staff       256 Apr 16 11:10 jre
> > drwxr-xr-x@ 14 erickchan  staff       448 Jul 10 18:02 lib
> > drwxr-xr-x@  5 erickchan  staff       160 Jun 28 14:51 resource
> 
> 14. 在Finder中運行"FFXIVChnTextPatchGP"
> 15. MacOS的安全檢測問"這個程式來自Internet，確認要執行嗎？"選擇"執行"。
> 16. 出現中文化程式界面，選擇"遊戲路徑"（請參考本Po上面樓主的解答選擇正確的遊戲路徑）
> 17. 檔案語言選擇"CSV"
> 18. 原始語言選擇"日文"
> 19. 點“確認”開始實行中文化

非常感謝大大，終於成功漢化了，可以好好享受劇情了！！

**Reply by OwenYangj** [reply]
請問我輸入完第二步驟 brew install openjdk 後出現 zsh: command not found: brew 
該如何安裝 openjdk 呢

**Reply by onlineid** [reply]
> 請問我輸入完第二步驟 brew install openjdk 後出現 zsh: command not found: brew 該如何安裝 openjdk 呢

請問第一步安裝brew成功了嗎？

**Reply by OwenYangj** [reply]
> 謝謝 改用echo加入SHELL後解決了

安裝完 brew 了，那個問題解決
想請問我遇到一樣的這個問題，如何用 echo 加入 shell 呢

**Reply by onlineid** [reply]
> > 謝謝 改用echo加入SHELL後解決了
> 
> 安裝完 brew 了，那個問題解決 想請問我遇到一樣的這個問題，如何用 echo 加入 shell 呢

請參考5-7

> 5. 把java路徑加入Shell環境變數
> echo "export PATH=\"/opt/homebrew/Cellar/openjdk/18.0.2/bin/:\$PATH\"" >> ~/.zshrc
> 
> 6. 確認5的操作結果
> cat ~/.zshrc
> export PATH="/opt/homebrew/Cellar/openjdk/18.0.2/bin/:$PATH"
> 
> 7. 令環境變數生效
> source ~/.zshrc

**Reply by OwenYangj** [reply]
感謝回覆，我最後執行 "FFXIVChnTextPatchGP" 時出現 The operation couldn’t be completed. Unable to locate a Java Runtime. 
是漏了哪個步驟呢？

**Reply by onlineid** [reply]
> 感謝回覆，我最後執行 "FFXIVChnTextPatchGP" 時出現 The operation couldn’t be completed. Unable to locate a Java Runtime.
> 是漏了哪個步驟呢？

5-7

**Reply by onlineid** [reply]
> 感謝回覆，我最後執行 "FFXIVChnTextPatchGP" 時出現 The operation couldn’t be completed. Unable to locate a Java Runtime.
> 是漏了哪個步驟呢？

請Po一下這兩條命令的輸出內容

which java
java --version

**Reply by OwenYangj** [reply]
@onlineid 抱歉我可能有很多新手問題，我有完成 5-7 步驟
8-13 也是要終端機嗎？還是我直接用 finder 解壓縮跟拉檔案就好？

**Reply by OwenYangj** [reply]
Last login: Sat Sep  3 21:48:44 on ttys001
j@JdeMacBook-Pro ~ % which java
/usr/bin/java
j@deMacBook-Pro ~ % java --version
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.

**Reply by onlineid** [reply]
> @onlineid 抱歉我可能有很多新手問題，我有完成 5-7 步驟
> 8-13 也是要終端機嗎？還是我直接用 finder 解壓縮跟拉檔案就好？

不會。我能回答的就回答。
8-13可以在Finder里做，不需要到終端機操作。

**Reply by onlineid** [reply]
> Last login: Sat Sep 3 21:48:44 on ttys001
> j@JdeMacBook-Pro ~ % which java
> /usr/bin/java
> j@deMacBook-Pro ~ % java --version
> The operation couldn’t be completed. Unable to locate a Java Runtime.
> Please visit http://www.java.com for information on installing Java.

你這邊的輸出顯示作業系統仍然使用Mac內建的Java。
你也是打算跟上面那位一樣用Mac內建的JDK嗎？那樣的話我不知道怎麼設定了。我是用的OpenJDK。
你可能要等上面那位來回答你的問題了。

**Reply by OwenYangj** [reply]
@onlineid  感恩，後來安裝 Java 就可以了

**Reply by HalseyHsieh** [reply]
請問我點了15之後出現下面這幾段後就沒反應了 是不是我漏了什麼
Last login: Wed Jan 11 00:14:44 on ttys000
/Users/halseyhsieh/Downloads/FFXIVChnTextPatch-1222/FFXIVChnTextPatchGP ; exit; 
halseyhsieh@HalseydeMacBook-Pro ~ % /Users/halseyhsieh/Downloads/FFXIVChnTextPatch-1222/FFXIVChnTextPatchGP ; exit;
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[程序完成]

**Comment by slayerkenneth** [comment]
@mosluce 大大你好 雖然我不是用你的release 我在intelliJ 也成功build了一樣PatchTool 但因為我用的是XIV On Mac, 我找不到FANTASY\ XIV\ONLINE/Bottles/published_Final_Fantasy/drive_c/Program\ Files\\(x86\)/SquareEnix/FINAL\ FANTASY\ XIV\ -\ A\ Realm\ Reborn 的這個路徑。不知道你有沒有甚麼解決方案呢
System: M1 Pro Macbook Pro
Platform: XIV On Mac (Windows License)

**Comment by mosluce** [comment]
XIV On Mac 應該是在

`/Users/[username]/Library/Application\ Support/XIV\ on\ Mac/ffxiv`

Kenneth Chan ***@***.***> 於 2022年9月21日 週三 凌晨12:17寫道：

**Reply by slayerkenneth** [reply]
找到了 感謝感謝！！！

**Comment by HalseyHsieh** [comment]
不好意思mac 新手 請問OpenJDK 檔案裡面沒看到 是要自行去下載嗎?
google 找不到你貼的三個OpenJDK

**Reply by onlineid** [reply]
> 不好意思mac 新手 請問OpenJDK 檔案裡面沒看到 是要自行去下載嗎? google 找不到你貼的三個OpenJDK

用Homebrew安裝。前面的回文有很詳細的方法。

**Reply by mosluce** [reply]
目前版本步驟已經可以大幅簡化
我現在都是這樣操作，提供給大家試試看
以下盡量提供指令可以複製貼上

以下都在 Terminal (終端機)，裡面操作，所以你至少要會打開終端機

## 準備

### 下載專案

```
# 做完此步驟，會在桌面多一個 FFXIVChnTextPatch-GP 資料夾
git clone https://github.com/GpointChen/FFXIVChnTextPatch-GP.git ~/Desktop/FFXIVChnTextPatch-GP
```

### 安裝 Homebrew

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

### 安裝 asdf

因作業系統版本不同會有不同的指令不要全部複製貼上

```
brew install coreutils curl git
brew install asdf

# === Catalina 以前的 macOS 專用 ===
echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ~/.bash_profile
source ~/.bash_profile
# ================================

# === Catalina 之後的 macOS 專用 ===
echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ${ZDOTDIR:-~}/.zshrc
source ~/.zshrc
# ================================

asdf plugin-add gradle https://github.com/rfrancis/asdf-gradle.git
asdf plugin-add java https://github.com/halcyon/asdf-java.git
```

### 安裝 openjdk

```
asdf install java openjdk-18
asdf global java openjdk-18
```

### 安裝 gradle

```
asdf install gradle 4.10.3
asdf global gradle 4.10.3
```

## 編輯 build.gradle

> 以下不是指令，打開 桌面/FFXIVChnTextPatch-GP/build.gradle 

找到 `maven { url "http://maven.aliyun.com/nexus/content/groups/public/" }`
改成 `maven { url "https://maven.aliyun.com/nexus/content/groups/public/" }`
有兩處

找到 `compile(files("libs/univocity-parsers-2.9.1.jar"))`
改成 `compile("com.univocity:univocity-parsers:2.9.1")`

## 執行

```
cd ~/Desktop/FFXIVChnTextPatch-GP
gradle run
```

**Reply by HalseyHsieh** [reply]
感謝大大的教學
小弟資質愚鈍
卡在 安裝asdf
不管貼哪一段都出現
zsh: command not found
或是
/Users/halseyhsieh/.zshrc:.:7: no such file or directory: /libexec/asdf.sh
是不是我的電腦少了某樣東西?

**Reply by mosluce** [reply]
@HalseyHsieh 
你有先跑 brew install asdf 嗎？

**Reply by mosluce** [reply]
@HalseyHsieh 先確定輸入
執行 `brew --prefix asdf` 會輸出 `/opt/homebrew/opt/asdf`

**Reply by mosluce** [reply]
@HalseyHsieh 確認
執行 `ls /opt/homebrew/opt/asdf/libexec` 會出現
![image](https://user-images.githubusercontent.com/1487051/212034950-b5165de6-92f0-40b1-b934-9cda5caffb78.png)

如果這邊正常 cmd+Q 把 Terminal 關掉重新打開，再看看是否還會出現錯誤
如果這邊沒有正常像上面顯示，表示前面兩個 brew install 你可能沒跑到

**Reply by HalseyHsieh** [reply]
感謝大大
<img width="487" alt="截圖 2023-01-12 下午8 37 13" src="https://user-images.githubusercontent.com/104269553/212068416-beccfc0f-d635-479a-8fa1-800a29686da3.png">
輸入還是只有顯示這反應
還是我輸入錯誤沒有指令之類的?

**Reply by onlineid** [reply]
似乎是沒安裝homebrew。
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

**Reply by HalseyHsieh** [reply]
應該是有的
<img width="567" alt="截圖 2023-01-12 下午9 12 54" src="https://user-images.githubusercontent.com/104269553/212075368-5b8f9a6c-2f80-464e-a2bb-0f0dca3b9b34.png">

**Reply by onlineid** [reply]
那可能是安裝不完整，最後三條指令沒有執行。

![halseyhsieh](https://user-images.githubusercontent.com/95579647/212078003-b4035a41-fb2a-42fe-83b9-bd59ed4125ab.jpg)

**Reply by HalseyHsieh** [reply]
感謝大大指點 我已經安裝好了
在接下來步驟下去 完成的畫面是不是這樣?

<img width="564" alt="截圖 2023-01-13 上午1 41 56" src="https://user-images.githubusercontent.com/104269553/212140096-09b235e7-488f-4fd8-a68b-7140c7a7a9ec.png">

**Reply by mosluce** [reply]
@HalseyHsieh 裝好 homebrew 

需要再次

```
brew install coreutils curl git
brew install asdf
```

把 asdf 裝起來

**Reply by mosluce** [reply]
@HalseyHsieh 我這裡打錯字

asdf global jave => asdf global java

<img width="618" alt="image" src="https://user-images.githubusercontent.com/1487051/212230870-83affa83-74fe-4a21-974c-ac71c72cec3b.png">

**Reply by mosluce** [reply]
@HalseyHsieh 另外這邊有誤解
<img width="604" alt="image" src="https://user-images.githubusercontent.com/1487051/212231078-edc016f3-e99c-4256-a779-337c870da435.png">

**Reply by mosluce** [reply]
@HalseyHsieh 你沒有安裝到 gradle

```
asdf install gradle 4.10.3
asdf global gradle 4.10.3
```

**Reply by HalseyHsieh** [reply]
可以了!!!!!!
感謝大大們的幫助!!!!!!

**Comment by mslauky** [comment]
卡在font6.tex 請問現在有解決方法了嗎？ 
爬文看見上面有其他人也有同樣情況出現
<img width="288" alt="螢幕截圖 2023-02-20 下午10 09 24" src="https://user-images.githubusercontent.com/62352923/220130025-2e89f7a6-b497-4f3a-b592-4878c3f9f3f6.png">


```
Exception in thread "Thread-1" java.lang.NoSuchMethodError: java.nio.ByteBuffer.position(I)Ljava/nio/ByteBuffer;
	at name.yumao.ffxiv.chn.builder.TexBlockBuilder.buildBlock(TexBlockBuilder.java:38)
	at name.yumao.ffxiv.chn.replace.ReplaceFont.replace(ReplaceFont.java:70)
	at name.yumao.ffxiv.chn.thread.ReplaceThread.run(ReplaceThread.java:46)
	at java.lang.Thread.run(Thread.java:750)
```

**Reply by william7928** [reply]
我也是卡在這 求解

**Reply by hydai** [reply]
Hi @william7928 
這個問題在新的討論串中解決了，可以嘗試看看： https://github.com/GpointChen/FFXIVChnTextPatch-GP/discussions/342

**Reply by william7928** [reply]
感謝解惑 將重返遊戲擁抱

**Comment by shiny10cean** [comment]
大家好，我是一個超廢但也想要中文化的mac使用者，請問有人能手把手帶我教學一下實際上應該怎麼執行嗎x口x
release 跟 build 都看不懂，超難執行... 會非常感謝...
（感覺ff14 的劇情就超好...

**Reply by GpointChen** [reply]
可以看看 https://github.com/GpointChen/FFXIVChnTextPatch-GP/discussions/342#discussion-5231768

**Reply by shiny10cean** [reply]
大大這回覆速度也太神速了！！！！！！！
馬上來試試！！！！！

**Reply by shiny10cean** [reply]
哇 一樣卡在這
<img width="289" alt="截圖 2023-09-13 下午6 57 47" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/102cd449-e7f9-4caf-a04b-dcb708f33462">

**Reply by shiny10cean** [reply]
看留言串有一位大大也是遇到一樣問題，你有重新上傳一個解決這問題的下載點但是剛剛看已經過期了

**Reply by GpointChen** [reply]
你是用 https://github.com/hydai/FFXIVChnTextPatch-GP/releases/tag/v0.0.8 這個版本嗎？

照理說下載對應的檔案並掛載.dmg後可以解決這個問題

你是說哪個下載點過期？

**Reply by shiny10cean** [reply]
啊 搞錯了是另一位大大上傳的，目前在等他回覆～

<img width="826" alt="截圖 2023-09-13 下午9 15 28" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/596ce36d-8426-4fec-8ece-e454c2477c18">

**Reply by GpointChen** [reply]
你可以點進我上面貼的那個v0.0.8的連結，裡面就有相關檔案可以下載，你按照他的安裝說明做即可。

**Reply by shiny10cean** [reply]
我目前是下載這個

FFXIVChnTextPatch-GP-AppleSilicon.dmg # <--- 如果你是 AppleSilicon(M1/M2/Mx) 版的 Mac
點兩下下載的檔案，掛載 dmg 以後，再點兩下 FFXIVChnTextPatchGP.app 執行即可

然後再 

下載 FFXIVChnTextPatch-GP-macOS.zip
解壓縮以後，點兩下 FFXIVChnTextPatchGP 執行即可

然後漢化之後就卡在剛剛的 font6.tex

**Reply by shiny10cean** [reply]
請問是也還要執行這些嗎
<img width="1215" alt="截圖 2023-09-13 下午9 31 32" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/9123d761-9162-41e5-b43a-b6256894cdd0">

**Reply by GpointChen** [reply]
不用，全部用他的即可

**Reply by GpointChen** [reply]
如果你還是會出現一樣的問題，那可能要等他確認了。

**Reply by shiny10cean** [reply]
了解好的！

**Reply by mslauky** [reply]
應該要等另一位大佬更新release了 因為他對應的版本不同 還是對應舊的 [PatchTool 6.48 8/13]

**Reply by shiny10cean** [reply]
嗚嗚 今天載了ff14 想說英文至少可以慢慢查翻譯，結果對話過一段時間就會消失，哭了直接沒搞

**Reply by shiny10cean** [reply]
該不會要直接去玩中國服吧 聽大家說都不推

**Reply by hydai** [reply]
同步到最新的 release 了，主要是之前看到 9/10 的更新只有修正一些翻譯問題就沒處理了XD

現在比較棘手的部分應該會是 JRE 在 macOS 上有分不同硬體版本，每次出版本時， apple silicon 的部分要我在本地端重新編譯一次
後續要研究一下怎麼讓 JRE 同時可以支援兩種硬體版本並且可以在 CI 上面自動化執行會比較方便

**Reply by mslauky** [reply]
@hydai 感謝大佬無私幫忙 mac玩家的確頭痛啊～

**Reply by shiny10cean** [reply]
嗚嗚嗚大大出現了啊！！！！

**Reply by shiny10cean** [reply]
附上一下我的java版本 （不知道有沒有用（哭
<img width="375" alt="截圖 2023-09-13 下午10 30 57" src="https://github.com/GpointChen/FFXIVChnTextPatch-GP/assets/144900232/ac484b3a-5b8c-4d75-b651-243f02d71374">
